### DISCUSSION TITLE
how do i patch leanback (flash)

### DISCUSSION THREAD

**Discussion by jackhacksren** [ftde0/yt2009#186] [Q&A] [answered]
using npapi adobe flash 32.0.0.371

with a browser of chrome 41

and even patching the players it somehow doesnt work

**Comment by jackhacksren** [comment]
patched my 2010 player and it now works for once

**Comment by jackhacksren** [comment]
i see no base url for it no guide bout it all

**Comment by jackhacksren** [comment]
it doesnt even play anything my flash player is 371 and using an chrome 41 browser

**Comment by ftde0** [comment, marked as answer]
follow the guide for 2006-2007 players for warp.

**Comment by jackhacksren** [comment]
it now works very thank you

**Comment by jackhacksren** [comment]
now how do i make leanback work tried to change the security domain allowing in the modulebase but it doesnt let me says not a type in line 21

**Comment by jackhacksren** [comment]
also what grep did you use to make the screenshot

**Comment by ftde0** [comment]
for leanback you should be able to get by by just patching the 2010 player in the leanback folder.

for grep i use [https://sourceforge.net/projects/grep-for-windows/files/](https://sourceforge.net/projects/grep-for-windows/files/), add the exe you extract to your PATH.

**Comment by jackhacksren** [comment]
tried patching the 2010 player in the leanback folder even with that patched it doesnt return any search queries

**Comment by jackhacksren** [comment]
now that i put all the base urls and security that

it now says an error has occured

and thats prob from the player site the search queries work just cant play

**Comment by jackhacksren** [comment]
![image_2023-12-22_143820294](https://github.com/ftde0/yt2009/assets/47057182/3db8b111-243f-42c4-89cb-eca758dd538a)

**Comment by jackhacksren** [comment]
[watch_as3-vfl183159.zip](https://github.com/ftde0/yt2009/files/13753021/watch_as3-vfl183159.zip)


take a look in that swf and see what i missed

**Comment by jackhacksren** [comment]
e

**Comment by ftde0** [comment]
you forgot a slash at the end of your `LIVE_BASE_URL` and `BASE_YT_URL` in YouTubeEnvironment.

```
public static const LIVE_BASE_URL:String = "http://127.0.0.1";
[...]
public var BASE_YT_URL:String = "http://127.0.0.1";
```
should be
```
public static const LIVE_BASE_URL:String = "http://127.0.0.1/";
[...]
public var BASE_YT_URL:String = "http://127.0.0.1/";
```

**Comment by jackhacksren** [comment]
now it sortaworks it played a video at the start but when i want to play my own videos it will fail

**Comment by jackhacksren** [comment]
and another thing even if it works it will only let me use one input of the arrow keys beforing forcing me to use the mouse

**Comment by ftde0** [comment]
> now it sortaworks it played a video at the start but when i want to play my own videos it will fail

mind linking the videos that don't play?

> and another thing even if it works it will only let me use one input of the arrow keys beforing forcing me to use the mouse

that's a problem of the leanback client itself, once you click with the arrows/mouse it will let you use arrow keys. won't fix

**Comment by jackhacksren** [comment]
> > now it sortaworks it played a video at the start but when i want to play my own videos it will fail
> 
> mind linking the videos that don't play?


it is any video that i want to play in the search queries

**Comment by jackhacksren** [comment]
how is it possible to host 192.168.1.4 cuz trying that i cant connect to it or see the webpage with that

**Comment by ftde0** [comment]
you need to change 192.168.1.4 as that used to be my local ip i've set for development. unless you set your local ip to be the same (which may not even work depending on your network settings) you will need to change it to your local ip.

**Comment by jackhacksren** [comment]
i set the hosting ip to be my local ip also its any video i play in the search queries
